src/etc/inc/plugins.inc.d/vxlan.inc

```diff
@@ -0,0 +1,130 @@
+        echo 'Configuring VXLAN interfaces...';
+        flush();
+    }
+    $all_addresses = array();
+    $known_addresses = array();
+    foreach ($interfaces_details as $intf) {
+        foreach (['ipv4', 'ipv6'] as $ipproto) {
+            if (!empty($intf[$ipproto])) {
+                foreach ($intf[$ipproto] as $net) {
+                    $known_addresses[] = $net['ipaddr'];
+                }
+            }
+        }
+    }
+
+    // (re)configure vxlan devices
+    foreach ($vxlans as $vxlan) {
+        if (!in_array((string)$vxlan->vxlanlocal, $known_addresses)) {
+            // skip when interface address is not assigned (yet)
+            continue;
+        }
+        $device_name = "vxlan.{$vxlan->deviceId}" ;
+        $configured_devices[] = $device_name;
+        $current_settings = array(
+            "vxlanid" => null,
+            "vxlanlocal" => null,
+            "vxlanremote" => null,
+            "vxlangroup" => null
+        );
+        if (empty($interfaces_details[$device_name])) {
+            // new device
+            mwexecf('/sbin/ifconfig vxlan create name %s', array($device_name));
+            $isChanged = true;
+        } else {
+            $isChanged = false;
+            $current_settings['vxlanid'] = $interfaces_details[$device_name]['vxlan']['vni'];
+            $current_settings['vxlanlocal'] = explode(":", $interfaces_details[$device_name]['vxlan']['local'])[0];
+            $current_settings['vxlanremote'] = explode(":", $interfaces_details[$device_name]['vxlan']['remote'])[0];
+            $current_settings['vxlangroup'] = explode(":", $interfaces_details[$device_name]['vxlan']['group'])[0];
+        }
+        // gather settings, detect changes
+        $ifcnfcmd = '/sbin/ifconfig %s';
+        $ifcnfcmdp = array($device_name);
+        foreach (array('vxlanid', 'vxlanlocal', 'vxlanremote', 'vxlangroup', 'vxlandev') as $param) {
+            $value = '';
+            if ($param == 'vxlandev') {
+                $intfnm = (string)$vxlan->$param;
+                if (!empty($cnf->interfaces->$intfnm)) {
+                    $value = $cnf->interfaces->$intfnm->if->__toString();
+                }
+            } else {
+                $value = (string)$vxlan->$param;
+            }
+            if ($value != '') {
+                $ifcnfcmd .= " {$param} %s ";
+                $ifcnfcmdp[] = $value;
+            }
+            if (isset($current_settings[$param]) && $current_settings[$param] != $value) {
+                $isChanged = true;
+            }
+        }
+        if ($isChanged) {
+            mwexecf($ifcnfcmd, $ifcnfcmdp);
+            $changed_devices[] = $device_name;
+        }
+    }
+    // destroy non existing interfaces
+    foreach ($interfaces_details as $intf => $data) {
+        if (strpos($intf, "vxlan.") === 0) {
+            if (!in_array($intf, $configured_devices)) {
+                mwexecf('/sbin/ifconfig %s destroy', array($intf));
```

Comment:
since I am sure this requires root -> configd

Reply:
it's within network plugin code, so we're good here.